include/oneapi/dpl/ranges

```diff
@@ -16,21 +16,29 @@
 #ifndef _ONEDPL_RANGES
 #define _ONEDPL_RANGES
 
+#if defined(_GLIBCXX_RELEASE)
```

Comment:
I see you don't include any standard library header. So, the `_GLIBCXX_RELEASE` macro shouldn't be defined. (See the example https://godbolt.org/z/vGebnv). I'd suggest the following:
```
// It should be done before any standard header inclusion
#if __has_include(<tbb/version.h>) 
#    ifndef PSTL_USE_PARALLEL_POLICIES
#        define PSTL_USE_PARALLEL_POLICIES (_GLIBCXX_RELEASE != 9)
#    endif
#    ifndef _GLIBCXX_USE_TBB_PAR_BACKEND
#        define _GLIBCXX_USE_TBB_PAR_BACKEND (_GLIBCXX_RELEASE > 10)
#    endif
#endif // __has_include(<tbb/version.h>)

#include<version>

#if defined(_GLIBCXX_RELEASE)
...
#endif
...
```

Reply:
Add TODO about the order of headers and inclusion of standard library headers version/ciso646.